opcotech/elemo
# feat: in-app notifications

## Description

This PR implements in-app notifications on both the backend and front-end.

## Supporting information

Closes https://github.com/opcotech/elemo/issues/135.

### Dependencies

N/A

### Screenshots

<img width="1735" alt="screenshot" src="https://github.com/opcotech/elemo/assets/19173947/d1df1076-9cfa-4573-86be-c685b1844d63">

## Testing instructions

1. Create a new notification manually in the DB
2. Flush the Redis cache
3. Start backend and front-end
4. Log in with Demo account
5. Check the notification indicator is shown
6. Ensure the notification shows when clicked on the bell icon

## Other information

The notifications are stored in the relational data store. Since this is the first first-party component that stores data there, the SQL DB bootstrapping is implemented in the `./scripts/setup.sh`.

## Checklist

  added `Signed-off-by: <YOUR NAME>` to the commit trail where `<YOUR NAME>` is
  my name.

### Documentation

- Documentation comments are updated
- Documentation site is updated

### Review

- Reviewed the code based on Go's [code review guide]
- Reviewed the code based on Go's [concurrency review guide]

### Miscellaneous

- Pull request is rebased onto main
- Commit history is clean



[Developer's Certificate of Origin]: https://github.com/opcotech/elemo/blob/main/CONTRIBUTING.md#developers-certificate-of-origin

[code review guide]: https://github.com/golang/go/wiki/CodeReviewComments

[concurrency review guide]: https://github.com/golang/go/wiki/CodeReviewConcurrency
